### ISSUE TITLE
New Rule: `prefer-global/timers`

### ISSUE THREAD

**Issue by benmccann** [eslint-community/eslint-plugin-n#339]
### Rule details

Similar to `prefer-global/process` it should enforce that `setImmediate` is imported from `node:timers`. This is needed in order to ensure Deno compatibility (https://github.com/denoland/deno/issues/17248#issuecomment-2325128835)

### What type of rule is this?

Warns about a potential problem

### Example code

```js
https://nodejs.org/api/timers.html#setimmediatecallback-args
```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

This probably applies to `setInterval` and `setTimeout` as well, which is why I proposed `prefer-global/timers` as a rule name
